relay-server/src/actors/processor.rs

```diff
@@ -1772,7 +1783,9 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState,
     ) -> Result<(), ProcessingError> {
-        if state.transaction_metrics_extracted {
+        if state.transaction_metrics_extracted
+            || state.early_enforcement.transaction_processed.is_active()
```

Comment:
Out of curiosity: If `transaction_processed` is active, doesn't it also mean that there is no `event` on the `state`, and we skip extracting on line 1809 anyway?